shell/browser/native_window_mac.mm

```diff
@@ -588,7 +588,14 @@ void ViewDidMoveToSuperview(NSView* self, SEL _cmd) {
 void NativeWindowMac::Close() {
   // When this is a sheet showing, performClose won't work.
   if (is_modal() && parent() && IsVisible()) {
-    [parent()->GetNativeWindow().GetNativeNSWindow() endSheet:window_];
+    NSWindow* window = parent()->GetNativeWindow().GetNativeNSWindow();
+    if (NSWindow* sheetParent = [window sheetParent]) {
+      base::ThreadTaskRunnerHandle::Get()->PostTask(
+          FROM_HERE, base::BindOnce(base::RetainBlock(^{
+            [sheetParent endSheet:window];
+          })));
+    }
+
     // Manually emit close event (not triggered from close fn)
     NotifyWindowCloseButtonClicked();
     CloseImmediately();
```

Comment:
The sheet is being closed async, will this function be safe to close the window synchronously ?

Reply:
@deepak1556 i believe so - this is the same approach that [chromium took](https://source.chromium.org/chromium/chromium/src/+/master:components/remote_cocoa/app_shim/views_nswindow_delegate.mm;l=137-149?q=%22%5BsheetParent%20endSheet:window%5D;%22&ss=chromium&originalUrl=https:%2F%2Fcs.chromium.org%2F)

Reply:
Thanks for the pointer.